test/jest/__mocks__/dexie.ts

```diff
@@ -0,0 +1,242 @@
+                dependencies: ["author-mod-1.0.0"],
+            }]
+        },
+        {
+            community: "RiskOfRain2",
+            full_name: "author-modWithTheSameDependency",
+            owner: "author",
+            name: "modWithTheSameDependency",
+            categories: [],
+            versions: [{
+                full_name: "modWithTheSameDependency",
+                name: "modWithTheSameDependency",
+                version_number: "2.0.0",
+                dependencies: ["author-mod-1.0.0"],
+            },{
+                full_name: "modWithTheSameDependency",
+                name: "modWithTheSameDependency",
+                version_number: "1.0.0",
+                dependencies: ["author-mod-1.0.0"],
+            }]
+        },
+        {
+            community: "RiskOfRain2",
+            full_name: "author-modWithTwoDependeciesThatHaveTheSameDependency",
+            owner: "author",
+            name: "modWithTwoDependeciesThatHaveTheSameDependency",
+            categories: [],
+            versions: [{
+                full_name: "modWithTwoDependeciesThatHaveTheSameDependency",
+                name: "modWithTwoDependeciesThatHaveTheSameDependency",
+                version_number: "2.0.0",
+                dependencies: ["author-modWithDependency-1.0.0", "author-modWithTheSameDependency-1.0.0"],
+            },{
+                full_name: "modWithTwoDependeciesThatHaveTheSameDependency",
+                name: "modWithTwoDependeciesThatHaveTheSameDependency",
+                version_number: "1.0.0",
+                dependencies: ["author-modWithDependency-1.0.0", "author-modWithTheSameDependency-1.0.0"],
+            }]
+        },
+        {
+            community: "RiskOfRain2",
+            full_name: "author-modPack",
+            owner: "author",
+            name: "modPack",
+            categories: [],
+            versions: [{
+                full_name: "modPack",
+                name: "modPack",
+                version_number: "2.0.0",
+                dependencies: ["author-mod-1.0.0"],
+            },{
+                full_name: "modPack",
+                name: "modPack",
+                version_number: "1.0.0",
+                dependencies: ["author-mod-1.0.0"],
+            }]
+        },
+        {
+            community: "RiskOfRain2",
+            full_name: "author-modWithDeepDependency",
+            owner: "author",
+            name: "modWithDeepDependency",
+            categories: [],
+            versions: [{
+                full_name: "modWithDeepDependency",
+                name: "modWithDeepDependency",
+                version_number: "2.0.0",
+                dependencies: ["author-modWithDependency-1.0.0"],
+            },{
+                full_name: "modWithDeepDependency",
+                name: "modWithDeepDependency",
+                version_number: "1.0.0",
+                dependencies: ["author-modWithDependency-1.0.0"],
```

Comment:
The test case where this is used would be slightly better if these two dependencies would depend on a different mod/version. So we could be certain that the dependency on the tail-end was picked from a correct "path".

Reply:
now it uses `modWithAnotherDependency` for the `1.0.0`